File: CS330/Assignment 4/Cylinder.cpp
#include "Cylinder.h"

//------------------------------------------------------------------------------
Cylinder::Cylinder()
    :Cylinder(1, 1)
{
}

//------------------------------------------------------------------------------
Cylinder::Cylinder(double r, double h)
    :Polyhedron("Cylinder"),
     height(h),
     radius(r)
{
    double d = this->getDiameter();
    boundingBox.setUpperRightVertex(d, d, height);
}

//------------------------------------------------------------------------------
void Cylinder::read(std::istream& ins)
{
    // Implement this function

    ins >> height >> radius;
    double d = this->getDiameter();
    boundingBox.setUpperRightVertex(d, d, height);
}

//------------------------------------------------------------------------------
void Cylinder::display(std::ostream& outs) const
{
    Polyhedron::display(outs);
    
    outs << "Radius: " << radius << " Height: " << height; 
}

//------------------------------------------------------------------------------
void Cylinder::scale(double scalingFactor)
{
    Polyhedron::scale(scalingFactor);
    // Implement this function
    radius *= scalingFactor;
    height *= scalingFactor;
}

Polyhedron* Cylinder::clone() const
{
    return new Cylinder(*this);
}

void Cylinder::setRadius(double r)
{
    radius = r;
    double d = this->getDiameter();
    boundingBox.setUpperRightVertex(d, d, height);
}

void Cylinder::setHeight(double h)
{
    height = h;
    double d = this->getDiameter();
    boundingBox.setUpperRightVertex(d, d, height);
}

File: CS330/Assignment 6/Cylinder.java
package polyhedra;

import java.util.Scanner;

/**
 * Polyhedron representing a cylinder.
 */
public class Cylinder extends Polyhedron {
    /**
     * Height of the cylinder. It must be > 0.
     */
    private double height;

    /**
     * Radius of the cylinder. It must be > 0.
     */
    private double radius;

    /**
     * Default Constructor - set the radius and height to 1.
     */
    public Cylinder()
    {
        this(1, 1);
    }

    /**
     * Construct a cylinder with specified height and radius.
     *
     * @param r desired radius
     * @param h deisred height
     */
    public Cylinder(double r, double h)
    {
        super("Cylinder");

        this.radius = r;
        this.height = h;

        double d = this.getDiameter();
        this.boundingBox.setUpperRightVertex(d, d, height);
    }

    /**
     * Retrieve the radius.
     *
     * @return current radius
     */
    public double getRadius()
    {
        return this.radius;
    }

    /**
     * Retrieve the height.
     *
     * @return current height
     */
    public double getHeight()
    {
        return this.height;
    }

    /**
     * Update the radius.
     *
     * @param r desired radius
     */
    public void setRadius(double r)
    {
        // Implement this function
        this.radius = r;

        double d = getDiameter();
        boundingBox.setUpperRightVertex(d, d, height);
    }

    /**
     * Update the height.
     *
     * @param h deisred height
     */
    public void setHeight(double h)
    {
        // Implement this function
        this.height = h;

        double d = getDiameter();
        boundingBox.setUpperRightVertex(d, d, height);
    }

    /**
     * Compute diameter.
     *
     * @return diameter
     */
    public final double getDiameter()
    {
        return this.radius * 2;
    }

    @Override
    public Polyhedron clone()
    {
        // remove this line
        // Implement this function
        return new Cylinder(this.radius, this.height);
    }

    @Override
    public void read(Scanner scanner)
    {
        // Implement this function
        this.height = scanner.nextDouble();
        this.radius = scanner.nextDouble();

        double d = this.getDiameter();
        boundingBox.setUpperRightVertex(d, d, height);
    }

    @Override
    public void scale(double scalingFactor)
    {
        // Implement this function
        this.height *= scalingFactor;
        this.radius *= scalingFactor;
        this.boundingBox.scale(scalingFactor);
    }

    @Override
    public String toString()
    {
                return super.toString()
             + "Radius: " + this.radius
             + " "
             + "Height: " + this.getHeight();
        //return String.format("height", this.height);
    }
}
